src/prefect/tasks/kubernetes/job.py

```diff
@@ -695,34 +709,76 @@ def run(
                 "The job name must be defined in the body under the metadata key."
             )
 
-        api_client = cast(
+        api_client_job = cast(
             client.BatchV1Api, get_kubernetes_client("job", kubernetes_api_key_secret)
         )
 
-        api_client.create_namespaced_job(namespace=namespace, body=body, **kube_kwargs)
+        api_client_pod = cast(
+            client.CoreV1Api, get_kubernetes_client("pod", kubernetes_api_key_secret)
+        )
+
+        api_client_job.create_namespaced_job(
+            namespace=namespace, body=body, **kube_kwargs
+        )
         self.logger.info(f"Job {job_name} has been created.")
 
-        completed = False
-        while not completed:
-            res = api_client.read_namespaced_job_status(
-                name=job_name, namespace=namespace
-            )
+        pool = ThreadPool()
+        pod_log_streams = dict()
+
+        try:
+            completed = False
+            while not completed:
+                job = api_client_job.read_namespaced_job_status(
+                    name=job_name, namespace=namespace
+                )
 
-            if res.status.active:
-                time.sleep(job_status_pool_interval)
-            else:
-                if res.status.failed:
+                if log_level is not None:
+                    func_log = getattr(self.logger, log_level)
+
+                    pod_selector = (
+                        f"controller-uid={job.metadata.labels['controller-uid']}"
+                    )
+                    pods_list = api_client_pod.list_namespaced_pod(
+                        namespace=namespace, label_selector=pod_selector
+                    )
+
+                    for pod in pods_list.items:
+                        pod_name = pod.metadata.name
+
+                        # Can't start logs when phase is pending
+                        if pod.status.phase == "Pending":
+                            continue
+                        if pod_name in pod_log_streams:
+                            continue
+
+                        read_pod_logs = ReadNamespacedPodLogs(
+                            pod_name=pod_name,
+                            namespace=namespace,
+                            kubernetes_api_key_secret=kubernetes_api_key_secret,
+                            func_stream=lambda log: func_log(f"{pod_name}: {log}"),
+                        )
+
+                        self.logger.info(f"Started following logs for {pod_name}")
+                        pod_log_streams[pod_name] = pool.apply_async(read_pod_logs.run)
```

Comment:
Eventually the thread pool will fill up, so active streaming of logs will only work for jobs with n-pods <= n-threads. This is a bit unfortunate - it'd be easy to support more if we were using an async kubernetes client but we're not.

I think I'm fine with this for now, but in the future I may want to fix this.

Reply:
Actually, would you mind adding a todo comment above this line with that info? Something like

```
# TODO: if a job has more active pods than the threadpool has threads
# logs will no longer be streamed live for all pods. Using an async
# k8s client would help remedy this.
```

Reply:
Aha alright, I'm not that well versed in Python multiprocessing, but good remark, I'll add it! An async kubernetes client would probably make this bit easier 😬 

To what size does Python's threadpool grow?

Reply:
> To what size does Python's threadpool grow?

It's static, and equal to the number of cores.

Actually we might want to switch to a [`concurrent.futures.TheadPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor) instead. This will grow in size but starts no threads by default (the default cap is `min(32, (os.cpu_count() or 1) + 4)`). You'd use `submit` instead of `apply_async` and `shutdown` instead of `close`, but everything else should be equivalent.